src/Compilers/Core/Portable/SourceGeneration/Nodes/IncrementalValueSources.cs

```diff
@@ -75,11 +87,13 @@ public InputNode<ISyntaxContextReceiver> GetOrCreateReceiverNode()
                 return InterlockedOperations.Initialize(ref _receiverNode, new InputNode<ISyntaxContextReceiver>());
             }
 
+            public ArrayBuilder<ISyntaxTransformNode> SyntaxTransformNodes { get; } = ArrayBuilder<ISyntaxTransformNode>.GetInstance();
+
             public PerGeneratorInputNodes ToImmutable()
```

Comment:
>ToImmutable [](http://example.com/codeflow?start=42&length=11)

If users of `Builder` are expected to call this method (to free `SyntaxTransformNodes`), consider calling `ToImmutableAndFree()` to indicate there are allocations to free.